meta_review/handler.py

```diff
@@ -0,0 +1,412 @@
+        """
+        self.logger = logging.getLogger(__name__)
+        self.logger.info('this package is alive')
+
+        self.date = date
+
+        # save comments in memory
+        self.comments = dict()
+        for issue in content:
+            issue = issue['issue']
+            if not issue.get('pullRequest'):
+                continue
+            for comment in issue['pullRequest']['comments']:
+                # parse time
+                comment['createdAt'] = parse_time(comment['createdAt'])
+                comment['lastEditedAt'] = parse_time(comment['lastEditedAt'])
+                self.comments[comment['id']] = comment
+
+        # save reactions in memory
+        self.reactions = dict()
+        for key, comment in self.comments.items():
+            if not comment.get('reactions'):
+                continue
+            for reaction in comment['reactions']:
+                # record receiver
+                reaction['receiver'] = {
+                    'login': comment['author']['login']
+                }
+                # record comment id
+                reaction['comment_id'] = comment['id']
+                # parse time
+                reaction['createdAt'] = parse_time(reaction['createdAt'])
+                self.reactions[reaction['id']] = reaction
+
+        # save participants in memory
+        self.participants = dict()
+        for key, comment in self.comments.items():
+            # get author of each comment
+            author = comment['author']
+
+            # skip if user does not exist
+            # this happens when account is deleted from GitHub
+            if author['login']:
+                self.participants[author['login']] = author
+
+            if not comment.get('reactions'):
+                continue
+            for reaction in comment['reactions']:
+                # get user of each reaction
+                user = reaction['user']
+
+                # skip if user does not exist
+                if author['login']:
+                    self.participants[user['login']] = user
+
+    def handle(self):
+        """
+        Scrape data, process and store in database
+        """
+        load_all(self.participants, self.comments, self.reactions)
+        dump_all(self.participants, self.comments, self.reactions)
+
+        self.__check_comment_update()
+        self.__update_score()
+        self.__update_weight_factors()
+        self.__update_time()
+
+        # dump first to make use of built-in sort method
+        dump_all(self.participants, self.comments, self.reactions)
+        self.__update_rankings()
+
+        dump_all(self.participants, self.comments, self.reactions)
+
+        self.logger.info('Meta Review System finishes.')
+
+    def __check_comment_update(self):
+        """
+        If reviewer updates their review comment after it has
+        been meta-reviewed, they will be slightly punished.
+        """
+        for key, comment in self.comments.items():
```

Comment:
"check_comment_update" isnt a good name for a method which modifies state based on complex logic.

The logic for determining who modified their review comments after meta-review should be separate from punishment.

It is very likely that other apps, especially the gamification app, might be interested in using the list of who is editing their comments.

Also the meta-review system might use it in multiple ways.  e.g. in the future, we might want to count how many times they are doing it, and log warnings for people who are regularly doing it, as the punishment isnt working.